src/beast/beast/asio/impl/buffers_readstream.ipp

```diff
@@ -0,0 +1,244 @@
+    {
+        return boost_asio_handler_alloc_helpers::
+            allocate(size, op->d_->h);
+    }
+
+    friend
+    auto asio_handler_deallocate(
+        void* p, std::size_t size, read_some_op* op)
+    {
+        return boost_asio_handler_alloc_helpers::
+            deallocate(p, size, op->d_->h);
+    }
+
+    friend
+    auto asio_handler_is_continuation(read_some_op* op)
+    {
+        return boost_asio_handler_cont_helpers::
+            is_continuation(op->d_->h);
+    }
+
+    template <class Function>
+    friend
+    auto asio_handler_invoke(Function&& f, read_some_op* op)
+    {
+        return boost_asio_handler_invoke_helpers::
+            invoke(f, op->d_->h);
+    }
+};
+
+template<class Stream, class ConstBufferSequence>
+template<class MutableBufferSequence, class Handler>
+void
+buffers_readstream<Stream, ConstBufferSequence>::
+read_some_op<MutableBufferSequence, Handler>::operator()(
+    error_code ec, std::size_t bytes_transferred)
+{
+    auto& d = *d_;
+    while(! ec && d.state != 99)
+    {
+        switch(d.state)
+        {
+        case 0:
+            if(d.brs.size_ == 0)
+            {
+                d.state = 1;
+                break;
+            }
+            d.state = 2;
+            d.brs.get_io_service().post(
+                bind_handler(std::move(*this), ec, 0));
+            return;
+
+        case 1:
+            // read
+            d.state = 99;
+            d.brs.next_layer_.async_read_some(
+                d.bs, std::move(*this));
+            return;
+
+        // copy
+        case 2:
+            bytes_transferred =
+                boost::asio::buffer_copy(
+                    d.bs, d.brs.cb_);
+            d.brs.size_ -= bytes_transferred;
+            d.brs.cb_.consume(bytes_transferred);
+            // call handler
+            d.state = 99;
+            break;
+        }
+    }
+    d.h(ec, bytes_transferred);
+}
+
+//------------------------------------------------------------------------------
+
+template<class Stream, class ConstBufferSequence>
+template<class... Args>
+buffers_readstream<Stream, ConstBufferSequence>::
+buffers_readstream(
+    ConstBufferSequence const& buffers, Args&&... args)
+    : size_(boost::asio::buffer_size(buffers))
+    , next_layer_(std::forward<Args>(args)...)
+    , cb_(buffers)
+{
+    static_assert(is_Stream<next_layer_type>::value,
+        "Stream requirements not met");
+    static_assert(
+        is_ConstBufferSequence<ConstBufferSequence>::value,
+            "ConstBufferSequence requirements not met");
+}
+
+template<class Stream, class ConstBufferSequence>
+template<class OtherConstBufferSequence, class WriteHandler>
+auto
+buffers_readstream<Stream, ConstBufferSequence>::
+async_write_some(OtherConstBufferSequence const& buffers,
+    WriteHandler&& handler)
+{
+    static_assert(is_ConstBufferSequence<
+        ConstBufferSequence>::value,
+            "ConstBufferSequence requirements not met");
```

Comment:
I suspect a copy/paste error here:  Was the intent to test `OtherConstBufferSequence` here instead of `ConstBufferSequence` (which has already been tested elsewhere)?